Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/NodePaletteModelHelper.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <EMotionFX/Source/AnimGraphNode.h>
+
+namespace GraphCanvas
+{
+    class NodePaletteWidget;
+    class NodePaletteTreeItem;
+}
+
+namespace EMStudio
+{
+    class AnimGraphPlugin;
+
+    /**
+     * This class build a hierarchy of GraphCanvas::NodePaletteTreeItem nodes
+     * that represent nodes that can be added to the anim graph when a given
+     * node is focused.
+     *
+     * Nodes will be sorted into categories.
+     *
+     * Root node returned by GetRootItem() is typically passed in a config
+     * for GraphCanvas::NodePaletteWidget setup. Object of this class should live
+     * no longer than the widget because the widget takes the ownership of the
+     * root node and the nodes in this class will become dangling pointers if the
+     * wigdet is destroyed.
+     */
```

Comment:
nit: Please now prefer `//!` style comments instead of `/**/` block comments for all API comments. They're much easier to maintain and less fiddly to format (repeated below)

Reply:
Done